### PR TITLE
Add ability for KerasClassifier to process string classes.

### PR DESCRIPTION
**Pull request by reiinakano**
This is a resubmission of #5321 

Scikit-learn classifiers are able to process strings in their target classes. This PR allows KerasClassifier to do the same while also exposing a `.classes_` attribute containing the classes used during fitting.

I modified the corresponding unit tests to test the behavior of KerasClassifier when fitted with string labels.

### CODE REVIEW

**Comment by fchollet** [comment]
@jdwittenauer @ipod825 could you please review this?

**Comment by jdwittenauer** [comment]
I'll try to take a look in the next couple days if no one else gets to it before then.

File: keras/wrappers/scikit_learn.py [hunk 1]
```diff
@@ -175,6 +175,27 @@ class KerasClassifier(BaseWrapper):
...
+            y : array-like, shape `(n_samples,)` or `(n_samples, n_outputs)`
+                True labels for X.
+            **kwargs: dictionary arguments
+                Legal arguments are the arguments of `Sequential.fit`
+
+        # Returns
+            history : object
+                details about the training history at each epoch.
+        """
+        self.classes_ = np.unique(y)
```
Comment (jdwittenauer) [inline keras/wrappers/scikit_learn.py]: This won't work correctly if y is already one-hot encoded right?  We need to check the shape of y first and handle appropriately.

File: tests/keras/wrappers/scikit_learn_test.py [hunk 2]
```diff
@@ -89,6 +92,24 @@ def assert_classification_works(clf):
     assert np.allclose(np.sum(proba, axis=1), np.ones(num_test))
 
 
+def assert_string_classification_works(clf):
+    str_y_train = np.array(['cls1', 'cls2', 'cls3'])[y_train]
```
Comment (jdwittenauer) [inline tests/keras/wrappers/scikit_learn_test.py]: This is too brittle IMO.  If someone changes num_class above then the test will start failing.  Try generating the string array based on num_class instead.

**Comment by reiinakano** [comment]
Hi @jdwittenauer , thanks for reviewing. I've addressed your comments in the new commit.

**Comment by jdwittenauer** [comment]
Looks good, thanks.  Think this can be merged.

**Comment by fchollet** [comment]
LGTM, thanks
